planet/src/main/webapp/js/placesItinerary.js

```diff
@@ -236,3 +245,32 @@ function validatePlaceDuration(duration) {
     }
     return true;
 }
+
+export function renderPlaceButtons(placeId, enableAdd) {
```

Comment:
If a place haven't been added, the button should be enabled, and it will be disabled as soon as the user adds the place.
Meanwhile, if the user deletes an event that was added from a place, the button should be enabled again.

> Example:

      user has a place called CN Tower --> user adds CN Tower as an event --> button disabled
                                       --> user deletes the CN Tower event --> button enabled

Reply:
It looks like we're putting several different functions into one here and controlling the code path with parameters. That makes it very difficult to read.

Q: What does renderPlaceButtons do?
A: Well, it renders all the places that the user has saved, except when it's called with a place id, in which case it only renders that place. It accepts an enableAdd boolean, but it only uses it if placeId is also set.

A function should do exactly one thing, which should be described by its name. Specifically to this case, this should probably be 3 functions:
disablePlaceButton(placeId)
enablePlaceButton(placeId)
renderPlaceButtons()

Reply:
That makes sense! Thank you :D